juju/juju
# Unit agent worker loggers

The purpose of this work is to pass loggers into the worker manifolds and remove the package level loggers. This work is particularly targeting the workers for the uniter. This work will mean when we merge the unit and machine agents the logs from the uniter workers will go to the right place.

There is a drive-by fix for the intermittent test failure in the featuretests package for the metrics watcher. The old attempt at fixing the failure with a WaitForModelWatchersIdle was not correct. The new fix has been tested under stress and stress-race loops locally.

Some worker tests were reworked a little so we didn't repeatedly have to add the loggers to many places, and instead makes use of suite methods.

## QA steps

Bootstrap and deploy and agent. If the engine manifolds all start up properly, then everything is wired up. Need to confirm with k8s and lxd to ensure we are covering all aspects of the manifolds.

## Documentation changes

Everything is internal, nothing to document.
